iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -227,7 +229,7 @@ class string
     /// @endcode
     // TruncateToCapacity_t is a compile time variable to distinguish between constructors
     // NOLINTNEXTLINE(hicpp-named-parameter, readability-named-parameter)
-    string(TruncateToCapacity_t, const char* const other) noexcept;
+    string(TruncateToCapacity_t, const char* const other) noexcept(false);
```

Comment:
Please use `noexcept` everywhere without `(false)`.

Reply:
@FerdinandSpitzschnueffler   I have removed all the  `(false)` used. There will be the another violations `A15-4-2` and `A15-5-3` are reported. I'm little unclear should suppress them. Could you suggest me ?

Reply:
@saif-at-github The other violations are all related to each other. I would propose to remove the `(false)` everywhere and ignore the related violations for now. We need to handle the problem that some functions can throw but are marked as `noexcept` probably in several places in our code base. So this could be part of a dedicated issue.